realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java

```diff
@@ -226,6 +231,129 @@ public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
         }
     }
 
+    // this builds a temporary schema that does not exist in io.realm.entities
+    private void buildBaseMigrationSchema(final String schemaName, final boolean createBase) {
+        Realm realm = Realm.getInstance(configFactory.createConfigurationBuilder().build());
+        realm.executeTransaction(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                // firstly remove an existing schema
+                realm.getSchema().remove(schemaName);
+                // then recreate the deleted schema or build base schema
+                realm.getSchema()
+                        .create(createBase ? MigrationPrimaryKey.CLASS_NAME : schemaName)
+                        .addField(MigrationPrimaryKey.FIELD_FIRST,   Byte.class)
+                        .addField(MigrationPrimaryKey.FIELD_SECOND,  Short.class)
+                        .addField(MigrationPrimaryKey.FIELD_PRIMARY, String.class, FieldAttribute.PRIMARY_KEY)
+                        .addField(MigrationPrimaryKey.FIELD_FOURTH,  Integer.class)
+                        .addField(MigrationPrimaryKey.FIELD_FIFTH,   Long.class);
+            }
+        });
+        realm.close();
+    }
+
+    @Test
+    public void renameTransferPrimaryKey() {
```

Comment:
Add a small comment about this, e..g : "Renaming the Primary Key field should also rename the field in the pk metadata table"